### ISSUE TITLE
VB.NET compiler erroneously reports error BC30439 when converting certain large Double values to Long

### ISSUE THREAD

**Issue by MichaelJLiu** [dotnet/roslyn#73032]
Any constant or literal `Double` should be convertible to `Long` at compile time if its value is within the range of the `Long` data type (approximately ±9.223E+18). However, if the value is a large positive number between `&H7000000000000400L` and `&H7FFFFFFFFFFFFC00L`, then the error `"BC30439: Constant expression not representable in type 'Long'."` occurs:
```vb
Console.WriteLine(CType(9.0E+18, Long))
'                       ~~~~~~~
Console.WriteLine(CType(CDbl(&H7000000000000400L), Long))
'                       ~~~~~~~~~~~~~~~~~~~~~~~~~
Console.WriteLine(CType(CDbl(&H7FFFFFFFFFFFFC00L), Long))
'                       ~~~~~~~~~~~~~~~~~~~~~~~~~

' Expected behavior: No errors.
' Actual behavior: Error BC30439 is reported on each line as indicated.
```

This bug does not affect `Double` values in the range `&H8000000000000000L` (`Long.MinValue`) through `&H7000000000000000L`:
```vb
Console.WriteLine(CType(-9.0E+18, Long))                  ' -9000000000000000000
Console.WriteLine(CType(CDbl(&H8000000000000000L), Long)) ' -9223372036854775808
Console.WriteLine(CType(CDbl(&H8000000000000400L), Long)) ' -9223372036854774784
Console.WriteLine(CType(CDbl(&H8FFFFFFFFFFFFC00L), Long)) ' -8070450532247929856
Console.WriteLine(CType(CDbl(&H9000000000000000L), Long)) ' -8070450532247928832
Console.WriteLine(CType(CDbl(&H7000000000000000L), Long)) '  8070450532247928832

' Expected and actual behavior: No errors.
```

This bug also does not affect run-time conversions:
```vb
Dim a = 9.0E+18
Dim b = CDbl(&H7000000000000400L)
Dim c = CDbl(&H7FFFFFFFFFFFFC00L)
Console.WriteLine(CType(a, Long)) ' 9000000000000000000
Console.WriteLine(CType(b, Long)) ' 8070450532247929856
Console.WriteLine(CType(c, Long)) ' 9223372036854774784

' Expected and actual behavior: No errors.
```

It looks like this bug affects all versions of the VB.NET compiler, including pre-Roslyn versions like C:\Windows\Microsoft.NET\Framework\v2.0.50727\vbc.exe.

This bug does not affect the C# compiler.
